tests/test_client_ws_functional.py

```diff
@@ -685,6 +685,36 @@ async def handler(request):
     assert resp.close_code is WSCloseCode.ABNORMAL_CLOSURE
 
 
+async def test_heartbeat_no_pong_concurrent_receive(aiohttp_client: Any) -> None:
+    ping_received = False
+
+    async def handler(request):
+        nonlocal ping_received
+        ws = web.WebSocketResponse(autoping=False)
+        await ws.prepare(request)
+        msg = await ws.receive()
+        if msg.type == aiohttp.WSMsgType.PING:
+            ping_received = True
+        ws._reader.feed_eof = lambda: None
+        await asyncio.sleep(10.0)
+        return ws
```

Comment:
Maybe we should raise here if its ever reached since that means something has gone wrong with the test and mark with no cover

Reply:
Perhaps, mark with `typing.assert_never()` and stick that into `exclude_also` in `.coveragerc`?

Reply:
If this is a question of coverage, we can exclude it. 
But I would draw your attention to the test this one is based on right above it in the code `test_heartbeat_no_pong` which I'm basing my code on, which has no assert as it's not a testable condition.

I would also kindly suggest that if there is a desire to refactor the tests overall, changing `==` or `is` and adding `asserts` then perhaps a separate issue/effort can be created for this. Refactoring one test leaving the other 20 _old style_ so to speak seems like trying to drain an ocean with a cup.

Reply:
I have fixed the line 700 not being covered by deleting it:

https://app.codecov.io/gh/aio-libs/aiohttp/pull/8546?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=aio-libs#68da58a6c1b4b9f2ac78950734ab927a-R699